### PR TITLE
[AMDGPU] Extend llvm.amdgcn.set.inactive intrinsic to support Reg32/Reg64 types

### PR DESCRIPTION
**Pull request by vikramRH**
Missed this while handling other patches. Any comments/concerns ?

### CODE REVIEW

File: llvm/test/CodeGen/AMDGPU/GlobalISel/llvm.amdgcn.set.inactive.ll [hunk 1]
```diff
@@ -93,6 +93,234 @@ define amdgpu_kernel void @set_inactive_scc(ptr addrspace(1) %out, i32 %in, <4 x
...
+; GCN-NEXT:    s_not_b64 exec, exec
+; GCN-NEXT:    s_mov_b32 s3, 0xf000
+; GCN-NEXT:    buffer_store_dword v0, off, s[0:3], 0
+; GCN-NEXT:    s_endpgm
+  %tmp = call <2 x i16> @llvm.amdgcn.set.inactive.v2i16(<2 x i16> %in, <2 x i16> <i16 1, i16 1>) #0
+  store <2 x i16> %tmp, ptr addrspace(1) %out
+  ret void
+}
+
+define amdgpu_kernel void @set_inactive_v2f16(ptr addrspace(1) %out, <2 x half> %in) {
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/GlobalISel/llvm.amdgcn.set.inactive.ll:160]: v4f16 is the other interesting case, plus bfloat

**Review by arsenm — COMMENTED** [review]
lgtm but I don't know if the graphics usage needs this

File: llvm/test/CodeGen/AMDGPU/GlobalISel/llvm.amdgcn.set.inactive.ll (hunk 1 above)
Comment (vikramRH) [reply llvm/test/CodeGen/AMDGPU/GlobalISel/llvm.amdgcn.set.inactive.ll:160]: added tests for v4i16, v4f16, v4bf16, v2bf16

**Comment by vikramRH** [comment]
> lgtm but I don't know if the graphics usage needs this

required for atomic optimizer for now, not sure about other use cases though

**Review by arsenm — APPROVED** [review]

**Pull request merged**